kork/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/artifactstore/entities/ManifestMapStorageHandler.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.kork.artifacts.artifactstore.entities;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.kork.artifacts.ArtifactTypeDecorator;
+import com.netflix.spinnaker.kork.artifacts.ArtifactTypes;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.ArtifactStore;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.filters.ApplicationFilter;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.filters.ApplicationStorageFilter;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * ManifestStorageHandler is used to convert manifest values into artifacts. This is done by
+ * matching against a key within the execution context. Upon matching, we will convert the manifest
+ * into an artifact and write the artifact in its place.
+ */
+public class ManifestMapStorageHandler implements ArtifactStorageHandler {
+  private List<String> keys = List.of("manifests", "outputs.manifests");
+  private final ApplicationFilter exclude;
+
+  public ManifestMapStorageHandler(Map<String, List<ApplicationStorageFilter>> exclude) {
+    this.exclude =
+        new ApplicationFilter(exclude.get(ArtifactTypes.EMBEDDED_MAP_BASE64.getMimeType()));
+  }
+
+  /** Manifest storage can only be handled on a list of maps. */
+  @Override
+  public boolean canHandle(Object v) {
+    if (AuthenticatedRequest.getSpinnakerExecutionId().isEmpty() || this.exclude.shouldFilter()) {
+      return false;
+    }
+
+    if (!(v instanceof Map)) {
+      return false;
+    }
+
+    Map m = (Map) v;
+    return this.keys.stream()
+        .anyMatch(
+            k -> {
+              Object o = m.get(k);
+              return isListOfManifests(o);
+            });
+  }
+
+  private static boolean isListOfManifests(Object v) {
+    if (!(v instanceof List)) {
+      return false;
+    }
+    List l = (List) v;
+    return !l.isEmpty() && isProperMapType(l.get(0));
+  }
+
+  private static boolean isProperMapType(Object o) {
+    if (!(o instanceof Map)) {
+      return false;
+    }
+
+    Map<?, ?> m = (Map<?, ?>) o;
+    if (m.isEmpty()) {
+      return false;
+    }
+    Map.Entry<?, ?> e = m.entrySet().stream().findFirst().get();
+    return e.getKey() instanceof String;
```

Comment:
Is this really specific enough to identify a manifest -- a map whose first key is a string?  Plus also, does `entrySet().stream()` guarantee order?  If the answer is no then this examines an arbitrary entry in the map.  I'd have expected this to look for something specific to kubernetes.  Can you add some comments explaining why this logic is the way it is?

Reply:
We are just type checking. Less about determining the contents of the manifests. If the keys are "manifests" or "outputs.manifests" we just make the assumption it's either remote or proper manifests

Reply:
> Am I right that:

    Map.Entry<?, ?> e = m.entrySet().stream().findFirst().get();
    return e.getKey() instanceof String;


> is the part that distinguishes a remote manifest from a proper one? Can you add a comment about that...and what non-String thing remote manifests are?

I dont think so. All this is doing is saying is the key a string. The `canHandle` method do not check anything about whether or not anything is remote or not.